File: java/iroha_android/src/main/java/org/hyperledger/iroha/android/client/ConfidentialAssetToriiException.java
package org.hyperledger.iroha.android.client;

/** Runtime error raised when a confidential-asset Torii request fails. */
public final class ConfidentialAssetToriiException extends RuntimeException {
  private static final long serialVersionUID = 1L;

  private final Integer statusCode;
  private final String rejectCode;
  private final String responseBody;

  public ConfidentialAssetToriiException(final String message, final Throwable cause) {
    this(message, cause, null, null, null);
  }

  public ConfidentialAssetToriiException(
      final String message,
      final Integer statusCode,
      final String rejectCode,
      final String responseBody) {
    this(message, null, statusCode, rejectCode, responseBody);
  }

  public ConfidentialAssetToriiException(
      final String message,
      final Throwable cause,
      final Integer statusCode,
      final String rejectCode,
      final String responseBody) {
    super(message, cause);
    this.statusCode = statusCode;
    this.rejectCode = blankToNull(rejectCode);
    this.responseBody = blankToNull(responseBody);
  }

  public Integer getStatusCode() {
    return statusCode;
  }

  public String getRejectCode() {
    return rejectCode;
  }

  public String getResponseBody() {
    return responseBody;
  }

  private static String blankToNull(final String value) {
    return value == null || value.trim().isEmpty() ? null : value;
  }
}

File: kotlin/core-jvm/src/main/java/org/hyperledger/iroha/sdk/client/ConfidentialAssetToriiException.kt
package org.hyperledger.iroha.sdk.client

/** Runtime error raised when a confidential-asset Torii request fails. */
class ConfidentialAssetToriiException : RuntimeException {
    val statusCode: Int?
    val rejectCode: String?
    val responseBody: String?

    constructor(message: String) : this(message, null, null, null, null)

    constructor(message: String, cause: Throwable?) : this(message, cause, null, null, null)

    constructor(
        message: String,
        statusCode: Int?,
        rejectCode: String?,
        responseBody: String?,
    ) : this(message, null, statusCode, rejectCode, responseBody)

    constructor(
        message: String,
        cause: Throwable?,
        statusCode: Int?,
        rejectCode: String?,
        responseBody: String?,
    ) : super(message, cause) {
        this.statusCode = statusCode
        this.rejectCode = rejectCode?.takeUnless(String::isBlank)
        this.responseBody = responseBody?.takeUnless(String::isBlank)
    }

    companion object {
        private const val serialVersionUID: Long = 1L
    }
}
